src/api/spec/components/notification_component_spec.rb

```diff
@@ -29,4 +29,35 @@
       expect(rendered_content).to have_css("i.fa-comments[title='Comment notification']")
     end
   end
+
+  context 'reports' do
+    let(:accused) { create(:confirmed_user, login: 'accused') }
+    let(:notification) { create(:notification_for_report, event_payload: event_payload, event_type: event_type, notifiable: notifiable) }
+
+    before do
+      render_inline(described_class.new(notification: notification, selected_filter: selected_filter, page: 1))
+    end
+
+    context 'when the notification is about user' do
+      let(:event_payload) { { reporter: user.login, reportable_id: accused.id, reportable_type: 'User', reason: 'some sample text for reason field', category: 'spam', accused: accused.login } }
+      let(:event_type) { 'Event::ReportForUser' }
+      let(:notifiable) { create(:report, reportable: accused, reason: 'Some sample text') }
+
+      it 'renders information about user state and its existing reports' do
```

Comment:
@rubhanazeem it only tests for the user state. Maybe you can test for the other badges too

Reply:
I've added a new assertion to check presence of other report/badge